refaster-support/src/test/java/tech/picnic/errorprone/refaster/matchers/AbstractMatcherTestChecker.java

```diff
@@ -34,27 +34,23 @@ abstract class AbstractMatcherTestChecker extends BugChecker implements Compilat
 
   @Override
   public Description matchCompilationUnit(CompilationUnitTree compilationUnit, VisitorState state) {
-    new TreeScanner<@Nullable Void, TreePath>() {
+    new TreePathScanner<@Nullable Void, @Nullable Void>() {
       @Override
-      public @Nullable Void scan(@Nullable Tree tree, TreePath treePath) {
-        if (tree == null) {
-          return null;
-        }
-
-        TreePath path = new TreePath(treePath, tree);
+      public @Nullable Void scan(@Nullable Tree tree, @Nullable Void unused) {
         if (tree instanceof ExpressionTree) {
+          TreePath path = new TreePath(getCurrentPath(), tree);
           ExpressionTree expressionTree = (ExpressionTree) tree;
           if (!isMethodSelect(expressionTree, path)
               && delegate.matches(expressionTree, state.withPath(path))) {
             state.reportMatch(describeMatch(tree));
           }
         }
 
-        return super.scan(tree, path);
+        return super.scan(tree, null);
       }
 
       @Override
-      public @Nullable Void visitImport(ImportTree tree, TreePath path) {
+      public @Nullable Void visitImport(ImportTree tree, @Nullable Void unused) {
         /*
          * We're not interested in matching import statements. While components of these
```

Comment:
What is the point of having an empty block in the visit method ?

Reply:
Always good to ask when in doubt! :+1: